src/dsp_tools/utils/excel_to_json/utils.py

```diff
@@ -0,0 +1,289 @@
+# pylint: disable=wrong-import-order
+
+from __future__ import annotations
+
+from typing import Any
+import numpy as np
+import pandas as pd
+import regex
+from unittest import mock
+
+from dsp_tools.models.exceptions import UserError
+
+languages = ["en", "de", "fr", "it", "rm"]
+
+
+def read_excel_file(excel_filename: str) -> pd.DataFrame:
+    """
+    This function reads an Excel file, if there is a ValueError then it patches the openpyxl part that creates the
+    error and opens it with that patch. It cleans and then returns the pd.DataFrame.
+
+    Args:
+        excel_filename: The name of the Excel file
+
+    Returns:
+        A pd.DataFrame
+    """
+    # load file
+    try:
+        read_df: pd.DataFrame = pd.read_excel(excel_filename)
+    except ValueError:
+        # Pandas relies on openpyxl to parse XLSX files.
+        # A strange behaviour of openpyxl prevents pandas from opening files with some formatting properties
+        # (unclear which formatting properties exactly).
+        # Apparently, the excel2json test files have one of the unsupported formatting properties.
+        # Credits: https://stackoverflow.com/a/70537454/14414188
+        with mock.patch("openpyxl.styles.fonts.Font.family.max", new=100):
+            read_df = pd.read_excel(excel_filename)
+    read_df = clean_data_frame(unclean_df=read_df)
+    return read_df
+
+
+def clean_data_frame(unclean_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    This function takes a pd.DataFrame and removes:
+        - Leading and trailing spaces from the column names
+        - Leading and trailing spaces from each cell
+        - Any characters in the cells that are not part of any known language, for example, linebreaks and replaces it
+          with a pd.NA.
+        - Removes all rows that are empty in all columns
+
+    Args:
+        unclean_df: The pd.DataFrame that is to be cleaned
+
+    Returns:
+        pd.DataFrame which has the above-mentioned removed
+    """
+    # Remove leading and trailing blanks in column names and make them lower case
+    cleaned_df = unclean_df.rename(columns=lambda x: x.strip().lower())
+    # Remove the values of all cells that do not at least contain one character of any known language
+    cleaned_df = cleaned_df.applymap(
+        lambda x: str(x).strip() if pd.notna(x) and regex.search(r"[\w\p{L}]", str(x), flags=regex.U) else pd.NA
+    )
+    # drop all the rows that are entirely empty
+    cleaned_df.dropna(axis=0, how="all", inplace=True)
+    return cleaned_df
+
+
+def check_required_columns_raise_error(check_df: pd.DataFrame, required_columns: set[str]) -> None:
+    """
+    This function takes a pd.DataFrame and a set of required column names. It checks if all the columns from the set
+    are in the pd.DataFrame. Additional columns to the ones in the set are allowed. It raises an error if any columns
+    are missing.
+
+    Args:
+        check_df: pd.DataFrame that is checked
```

Comment:
I would rename `check_df` to `df_to_be_checked` (or similar)

Reply:
I would argue to simply call it `df`: the variable has a lifetime of only 4 lines of code and it's the first argument of a function called `check_XY`. Similarly, `required_columns` could just be `columns`, if the function is called `check_required_columns(df, columns)` it's still perfectly clear what's going on

Reply:
In the case of columns I understand your point. But with the df you can get bugs that are annoying to find. Because if you do a change in memory in a df then it changes it everywhere... 
<img width="1206" alt="Screenshot 2023-08-30 at 10 22 55" src="https://github.com/dasch-swiss/dsp-tools/assets/103038637/3b15ec79-d47c-4005-a0b8-b8a10a443e90">

But maybe it would be worth a discussion to remove all inplace where there is no performance improvement (https://sourcery.ai/blog/pandas-inplace/)?

Reply:
Are you sure this problem is tied to variable names? I think the problem is the inplace, not how the variables are named.

Reply:
True, you are correct. Probably I resolved it by removing the inplace in the past (it was a while ago). But generally I also tend to add information to the variables as in the past I often worked with several different dfs at the same time and it makes keyword arguments (which I prefer) clearer to me...

Reply:
Me too, I prefer keyword arguments 😄. 

I think the reason I dislike `check_df` is that in natural language, it sounds like a command ("Hey computer, please check this df"), which would be a good function name, but a bad variable name. That's why I prefer `df_to_be_checked`, because in natural language, it's "a dataframe that needs to be checked".

But actually, I prefer Balduin's input over my initial input: The variable only lives during 4 lines of code. And every function is a separate scope, so if so set a breakpoint here, and execution halts here, you only see the variables of this scope (=this function). So I don't see a potential to confuse it with other variables, if it's just named `df`

Reply:
1) 
   a) I think if we prefer shortness over understandability, we don't do a favor to other developers and to our future selves :-) 
   b) For me, verbs are very much tied to function names, because functions **do** something. So, a dataframe should not have a verb in its variable name.

2) I think it really depends on the context. In a short function with only 1 dataframe, I prefer `df`. But in a complex function with multiple dataframes I prefer longer names. I don't think this is inconsistent. Because it has to make sense in the context

But now we have discussed a lot about a single variable name 😂  . I like such discussions, because we can learn something, but at some point, we should just take a decision. I leave this decision to you 😄

Reply:
It is good to have the conversation now i think